lightning/src/ln/peer_handler.rs

```diff
@@ -613,7 +582,34 @@ impl<Descriptor: SocketDescriptor, CM: Deref> PeerManager<Descriptor, CM> where
 										peer.pending_read_is_header = true;
 
 										let mut reader = ::std::io::Cursor::new(&msg_data[..]);
-										let message = try_potential_decodeerror!(wire::read(&mut reader));
+										let message_result = wire::read(&mut reader);
+										let message = match message_result {
```

Comment:
You can simplify these two lines as:

```rust
let message = match wire::read(&mut reader) {
```

Or use the name `message` in both assignments to shadow the first variable, as that is idiomatic in Rust.

Reply:
True. Although I would argue it might be easier for readers and debuggers to have the match input be a variable as opposed to a method call. What do you think?

Reply:
I don't have a strong opinion on that. For the readability standpoint, the `let x = match <exp> { Ok(x) => x, Err(e) => { ... } }` pattern is common enough to know that the expression evaluates to a `Result`. Regarding debugging, knowing which arm is hit should be enough.

Anyhow, feel free to leave as is if you feel strongly.